### DISCUSSION TITLE
Font rendering differences

### DISCUSSION THREAD

**Discussion by sbaildon** [kovidgoyal/kitty#4485] [Q&A]
I'm using Apple's SF Mono as my terminal font; kitty's rendering is different from Apple's, and I'm wondering if I can get parity with `Terminal.app`

```
~ $ kitty +list-fonts --psnames

SF Mono
    SF Mono Medium (SFMono-Medium)
    SF Mono Regular (SFMono-Regular)
```

Here's `Terminal.app` with SF Mono Regular, 11
<img width="633" alt="image" src="https://user-images.githubusercontent.com/214264/148677309-c35e938c-3820-4019-8c6d-9192c497ca91.png">

Here's `kitty` with SF Mono Regular, 11 (`kitty -o "font_family=SFMono-Regular"`)
<img width="623" alt="image" src="https://user-images.githubusercontent.com/214264/148677386-63104b0d-e696-4d55-9880-d8f3b322805d.png">

And for posterity, here's `TextEdit.app` with font rendering matching Apple's own `Terminal.app`

<img width="622" alt="image" src="https://user-images.githubusercontent.com/214264/148677600-2ec5c615-baac-4d07-9bf4-c4e4f56f3216.png">

I've toyed with `macos_thicken_font`, but it's certainly not as sharp as Apple's rendering—notice the feet of `1` and `2`, and the stem in the `b` 

`Terminal.app` reference

<img width="688" alt="image" src="https://user-images.githubusercontent.com/214264/148677931-f2b3559d-1321-44ee-9abe-5fffb8a424c7.png">

`kitty` reference (`kitty -o "font_family=SFMono-Regular" -o "font_size=11" -o "macos_thicken_font=0.4"`)

<img width="678" alt="image" src="https://user-images.githubusercontent.com/214264/148677949-edba1026-a7e4-4d2f-b792-8d1923c888c2.png">

**Comment by page-down** [comment]
Using `macos_thicken_font` only increases the width of the text strokes, instead it becomes a little bit blurry.
It cannot increase the sharpness.

I noticed this problem when I started using kitty for the first time, and still don't know how to adjust CoreText / CoreGraphics / OpenGL GLSL to achieve the similar rendering result.

The following PR is also related to text rendering
https://github.com/kovidgoyal/kitty/pull/3308
